### PR TITLE
Fix aliases with positional args enabled

### LINKED ISSUE #31173: enable_positional_arguments doesnt work with substr(key, 10)
**Describe what's wrong**

ClickHouse doesn't understand that argument belongs to substr function

**Does it reproduce on recent release?**

21.11

**How to reproduce**
```
WITH 'aaaaaaaaaaaaaa' AS key
SELECT substr(key, 10) AS key_a
FROM numbers(10)
GROUP BY key_a
SETTINGS enable_positional_arguments = 1


0 rows in set. Elapsed: 0.002 sec.

Received exception from server (version 21.12.1):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Positional argument out of bounds: 10 (exprected in range [1, 1]. (ILLEGAL_TYPE_OF_ARGUMENT)
```

**Expected behavior**

Query works

**Error message and/or stacktrace**

Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Positional argument out of bounds: 10 (exprected in range [1, 1]. (ILLEGAL_TYPE_OF_ARGUMENT)

### PR DESCRIPTION
**Pull request by kssenii**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix group by / order by / limit by aliases with positional arguments enabled. Closes https://github.com/ClickHouse/ClickHouse/issues/31173.

### CODE REVIEW

File: src/Interpreters/ExpressionAnalyzer.cpp [hunk 1]
```diff
@@ -155,12 +170,7 @@ bool checkPositionalArguments(ASTPtr & argument, const ASTSelectQuery * select_q
                 }
             }
             else if (pos > columns.size() || !pos)
-            {
-                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
```
Comment (novikd) [inline src/Interpreters/ExpressionAnalyzer.cpp:159]: As I understand now we will ignore incorrect positional arguments?

Comment (kssenii) [reply src/Interpreters/ExpressionAnalyzer.cpp:159]: If out of bound.

Comment (kssenii) [reply src/Interpreters/ExpressionAnalyzer.cpp:159]: ok

**Comment by kssenii** [comment]
@Mergifyio update

**Comment by kssenii** [comment]
Integration tests (asan, actions) — fail: 8 -- failed because of docker issues
```
1. test_backward_compatibility
>               raise Exception(f"Command {args} return non-zero code {res.returncode}: {res.stderr.decode('utf-8')}")
E               Exception: Command ['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_backward_compatibility/_instances_skipping_indices_0/.env', '--project-name', 'roottestbackwardcompatibilityskippingindices', '--file', '/ClickHouse/tests/integration/test_backward_compatibility/_instances_skipping_indices_0/node/docker-compose.yml', 'up', '-d', '--no-recreate'] return non-zero code 1: Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/

2. test_storage_url and test_disk_over_web_server
>               raise Exception(f"Command {args} return non-zero code {res.returncode}: {res.stderr.decode('utf-8')}")
E               Exception: Command ['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_storage_url/_instances_0/.env', '--project-name', 'rootteststorageurl', '--file', '/compose/docker_compose_nginx.yml', '--verbose', 'up', '-d', '--renew-anon-volumes'] return non-zero code 1: compose.config.config.find: Using configuration files: /compose/docker_compose_nginx.yml
```
01154_move_partition_long -- known flacky test

**Review by novikd — APPROVED** [review]

**Pull request merged**